### PR TITLE
UI: Clean up deprecated presentation-related APIs

### PR DESCRIPTION
**Pull request by grigasp**
- Remove the deprecated `Tree` component.
- Remove drag & drop APIs.
- Remove deprecated version of `FilteringInput` component props.
- Make `width` and `height` required for virtualized property grid.

### CODE REVIEW

**Review by pmconne — APPROVED** [review]

**Review by DanEastBentley — APPROVED** [review]

**Review by roluk — CHANGES_REQUESTED** [review]

File: test-apps/presentation-test-app/src/frontend/components/properties-widget/PropertiesWidget.tsx [hunk 1]
```diff
@@ -170,6 +177,8 @@ function PropertyGrid(props: PropertyGridProps) {
 
   return <>
     <VirtualizedPropertyGridWithDataProvider
+      width={width}
+      height={height}
       dataProvider={filteringDataProvider}
       isPropertyHoverEnabled={true}
       onPropertyContextMenu={onPropertyContextMenu}
```
Comment (roluk) [inline test-apps/presentation-test-app/src/frontend/components/properties-widget/PropertiesWidget.tsx:184]: Property Grid in presentation-test-app no longer switches to vertical mode when it crosses the width threshold.

**Review by roluk — CHANGES_REQUESTED** [review]

File: ui/components/src/test/properties/renderers/value/WithContextStyle.test.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+*--------------------------------------------------------------------------------------------*/
+import { expect } from "chai";
+import { mount } from "enzyme";
+import * as React from "react";
+import { withContextStyle } from "../../../../ui-components/properties/renderers/value/WithContextStyle";
+
+describe("withContextStyle", () => {
+
+  it("returns given node when context is not provided", () => {
+    const reactNode: React.ReactNode = (<>test</>);
```
[the remark this answers is not in the document]
Comment (grigasp) [reply ui/components/src/test/properties/renderers/value/WithContextStyle.test.tsx]: fixed

File: docs/changehistory/NextVersion.md [hunk 3]
```diff
@@ -319,19 +319,28 @@ SAML support has officially been dropped as a supported workflow. All related AP
...
+| `Timeline`                                                  | `TimelineComponent` in @bentley/ui-components                         |
+| `ControlledTreeProps.treeEvents`                            | `ControlledTreeProps.eventsHandler`                                   |
+| `ControlledTreeProps.visibleNodes`                          | `ControlledTreeProps.model`                                           |
+| `MutableTreeModel.computeVisibleNodes`                      | `computeVisibleNodes` in @bentley/ui-components                       |
+| `TreeModelSource.getVisibleNodes`                           | memoized result of `computeVisibleNodes`                              |
+| `useVisibleTreeNodes`                                       | `useTreeModel` and `computeVisibleNodes`                              |
+| `SignIn`                                                    | *eliminated*                                                          |
+| All drag & drop related APIs                                | Third party components. E.g. see this [example](https://www.itwinjs.org/sample-showcase/?group=UI+Trees&sample=drag-and-drop) |
```
[the remark this answers is not in the document]
Comment (grigasp) [reply docs/changehistory/NextVersion.md:346]: fixed

File: test-apps/presentation-test-app/src/frontend/components/properties-widget/PropertiesWidget.tsx (hunk 1 above)
Comment (grigasp) [reply test-apps/presentation-test-app/src/frontend/components/properties-widget/PropertiesWidget.tsx:184]: fixed

File: docs/changehistory/NextVersion.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (roluk) [reply docs/changehistory/NextVersion.md:346]: There are still some changes when I auto-format this document. Are you formatting markdown by hand?

File: ui/components/src/test/properties/renderers/value/WithContextStyle.test.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (roluk) [reply ui/components/src/test/properties/renderers/value/WithContextStyle.test.tsx]: Thanks

File: test-apps/presentation-test-app/src/frontend/components/properties-widget/PropertiesWidget.tsx (hunk 1 above)
Comment (roluk) [reply test-apps/presentation-test-app/src/frontend/components/properties-widget/PropertiesWidget.tsx:184]: Thanks

File: docs/changehistory/NextVersion.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (grigasp) [reply docs/changehistory/NextVersion.md:346]: yes, I'm not formatting stuff others' stuff

[the remark this answers is not in the document]
Comment (roluk) [reply docs/changehistory/NextVersion.md:346]: It's your changes that are still not properly formatted. Please install [Markdown All in One](https://marketplace.visualstudio.com/items?itemName=yzhang.markdown-all-in-one) and auto-format the document.

[the remark this answers is not in the document]
Comment (grigasp) [reply docs/changehistory/NextVersion.md:346]: should be good now

**Review by roluk — APPROVED** [review]

**Pull request merged**